File: mobile/src/main/java/at/shockbytes/remote/network/message/JsonMessageSerializer.java
package at.shockbytes.remote.network.message;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Martin Macheiner
 *         Date: 27.09.2017.
 */

public class JsonMessageSerializer implements MessageSerializer {

    @Override
    public String mouseMoveMessage(int deltaX, int deltaY) {
        JSONObject object = new JSONObject();
        try {
            object.put("deltaX", deltaX);
            object.put("deltaY", deltaY);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return object.toString();
    }

    @Override
    public String writeTextMessage(int keyCode, boolean isUpperCase) {
        JSONObject object = new JSONObject();
        try {
            object.put("keycode", keyCode);
            object.put("uppercase", isUpperCase);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return object.toString();
    }

    @Override
    public String keyExchange(String deviceName, String certificate) {
        JSONObject object = new JSONObject();
        try {
            object.put("device", deviceName);
            object.put("certificate", certificate);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return object.toString();    }
}

File: mobile/src/main/java/at/shockbytes/remote/network/message/JsonMessageSerializer.kt
package at.shockbytes.remote.network.message

import com.google.gson.JsonObject

/**
 * @author Martin Macheiner
 * Date: 27.09.2017.
 */

class JsonMessageSerializer : MessageSerializer {

    override fun mouseMoveMessage(deltaX: Int, deltaY: Int): String {
        val element = JsonObject()
        element.addProperty("deltaX", deltaX)
        element.addProperty("deltaY", deltaY)
        return element.toString()
    }

    override fun writeTextMessage(keyCode: Int, isUpperCase: Boolean): String {
        val element = JsonObject()
        element.addProperty("keycode", keyCode)
        element.addProperty("uppercase", isUpperCase)
        return element.toString()
    }

    override fun keyExchange(deviceName: String, certificate: String): String {
        val element = JsonObject()
        element.addProperty("device", deviceName)
        element.addProperty("certificate", certificate)
        return element.toString()
    }
}
